nrpy/infrastructures/BHaH/MoLtimestepping/tests/MoL_MoL_step_forward_in_time.c

```diff
@@ -1,5 +1,117 @@
 #include "BHaH_defines.h"
 #include "BHaH_function_prototypes.h"
+/**
+ * Runge-Kutta function for substep 1.
+ */
+static void rk_substep_1(params_struct *restrict params, REAL *restrict k_odd_gfs, REAL *restrict y_n_gfs, REAL *restrict y_nplus1_running_total_gfs,
+                         const REAL dt) {
+  const int Nxx_plus_2NGHOSTS0 = params->Nxx_plus_2NGHOSTS0;
+  const int Nxx_plus_2NGHOSTS1 = params->Nxx_plus_2NGHOSTS1;
+  const int Nxx_plus_2NGHOSTS2 = params->Nxx_plus_2NGHOSTS2;
+  const int Ntot = Nxx_plus_2NGHOSTS0 * Nxx_plus_2NGHOSTS1 * Nxx_plus_2NGHOSTS2 * NUM_EVOL_GFS;
+
+  LOOP_ALL_GFS_GPS(i) {
+    const REAL k_odd_gfsL = k_odd_gfs[i];
+    const REAL y_n_gfsL = y_n_gfs[i];
+    const REAL RK_Rational_1_6 = 1.0 / 6.0;
+    const REAL RK_Rational_1_2 = 1.0 / 2.0;
+    y_nplus1_running_total_gfs[i] = RK_Rational_1_6 * dt * k_odd_gfsL;
+    k_odd_gfs[i] = RK_Rational_1_2 * dt * k_odd_gfsL + y_n_gfsL;
+  }
+} // END FUNCTION rk_substep_1
+
+/**
+ * Runge-Kutta function for substep 1.
+ */
+static void rk_substep_1__launcher(params_struct *restrict params, REAL *restrict k_odd_gfs, REAL *restrict y_n_gfs,
+                                   REAL *restrict y_nplus1_running_total_gfs, const REAL dt) {
+  rk_substep_1(params, k_odd_gfs, y_n_gfs, y_nplus1_running_total_gfs, dt);
+} // END FUNCTION rk_substep_1__launcher
+
+/**
+ * Runge-Kutta function for substep 2.
+ */
+static void rk_substep_2(params_struct *restrict params, REAL *restrict k_even_gfs, REAL *restrict y_nplus1_running_total_gfs, REAL *restrict y_n_gfs,
+                         const REAL dt) {
+  const int Nxx_plus_2NGHOSTS0 = params->Nxx_plus_2NGHOSTS0;
+  const int Nxx_plus_2NGHOSTS1 = params->Nxx_plus_2NGHOSTS1;
+  const int Nxx_plus_2NGHOSTS2 = params->Nxx_plus_2NGHOSTS2;
+  const int Ntot = Nxx_plus_2NGHOSTS0 * Nxx_plus_2NGHOSTS1 * Nxx_plus_2NGHOSTS2 * NUM_EVOL_GFS;
+
+  LOOP_ALL_GFS_GPS(i) {
+    const REAL k_even_gfsL = k_even_gfs[i];
+    const REAL y_nplus1_running_total_gfsL = y_nplus1_running_total_gfs[i];
+    const REAL y_n_gfsL = y_n_gfs[i];
+    const REAL RK_Rational_1_3 = 1.0 / 3.0;
```

Comment:
Let's make these rationals `static const` instead. Most compilers would optimize it that way, but not all with all compile flags.